main.go

```diff
@@ -114,18 +139,44 @@ func runMain() error {
 	}
 }
 
-func uploadFiles(apiClient *api.CacophonyAPI, directory string) error {
+func uploadFiles(apiClient *api.CacophonyAPI, directory string) (bool, error) {
 	matches, _ := filepath.Glob(filepath.Join(directory, cptvGlob))
+	success := true
+	var err error
 	for _, filename := range matches {
-		err := uploadFileWithRetries(apiClient, filename)
+		success, err = uploadFileWithRetries(apiClient, filename)
 		if err != nil {
-			return err
+			return success, err
+		}
+	}
+	return success, nil
+}
+
+func retryFailedUploads(apiClient *api.CacophonyAPI, directory string) bool {
+	matches, _ := filepath.Glob(filepath.Join(directory, failedUploadsDir, cptvGlob))
+	if len(matches) == 0 {
+		return true
+	}
+	// start at a random index to avoid always failing on the same file
+	startIndex := rand.Intn(len(matches))
+	for i := 0; i < len(matches); i++ {
+		index := (startIndex + i) % len(matches)
+		filename := matches[index]
+		err := uploadFile(apiClient, filename)
+		if err != nil {
+			log.Printf("failed uploading failed recording %v: %v", filename, err)
+			return false
+		}
+		log.Printf("success uploading failed recording\n")
```

Comment:
This can just be `log.Print`. Also: no need for the \n as newlines get added automatically by the logging functions anyway.